### DISCUSSION TITLE
`streamingProcess: exec: invalid argument (Bad file descriptor)` when examining ghc source code

### DISCUSSION THREAD

**Discussion by k355l3r-5yndr0m3** [mrcjkb/haskell-tools.nvim#335] [Q&A]
### Neovim version (nvim -v)

v0.9.5

### Operating system/version

Archlinux 

### Output of :checkhealth haskell-tools

```console
==============================================================================
haskell-tools: require("haskell-tools.health").check()

Checking for Lua dependencies ~
- WARNING telescope not installed. Required for hoogle search modes "telescope-local" and "telescope-web" [nvim-telescope/telescope.nvim](https://github.com/nvim-telescope/telescope.nvim)

Checking external dependencies ~
- OK haskell-language-server: found haskell-language-server version: 2.5.0.0 (GHC: 9.2.8) (PATH: /home/cha0s/.ghcup/hls/2.5.0.0/lib/haskell-language-server-2.5.0.0/bin/haskell-language-server-wrapper)
- WARNING       hoogle: not found.
  Install [ndmitchell/hoogle](https://github.com/ndmitchell/hoogle) for extended capabilities.
  Recommended for better Hoogle search performance.
  Without a local installation, the web API will be used by default.
  Required if the hoogle mode is set to "telescope-local".
  
  
- WARNING       fast-tags: not found.
  Install [fast-tags](https://hackage.haskell.org/package/fast-tags) for extended capabilities.
  Optional, for generating tags as a `tagfunc` fallback.
  
- OK curl: found curl 8.5.0 (x86_64-pc-linux-gnu) libcurl/8.5.0 OpenSSL/3.2.0 zlib/1.3 brotli/1.1.0 zstd/1.5.5 libidn2/2.3.4 libpsl/0.21.2 (+libidn2/2.3.4) libssh2/1.11.0 nghttp2/1.58.0
- WARNING       haskell-debug-adapter: not found.
  Install [haskell-debug-adapter](https://github.com/phoityne/haskell-debug-adapter) for extended capabilities.
  Optional, for `dap` support.
  
- WARNING       ghci-dap: not found.
  Install [ghci-dap](https://github.com/phoityne/ghci-dap) for extended capabilities.
  Optional, for `dap` support.
  

Checking config ~
- OK No errors found in config.

Checking for conflicting plugins ~
- OK No conflicting plugins detected.
```

### How to reproduce the issue

default config

### Expected behaviour

auto complete, linting

### Actual behaviour

`ghc/compiler/hadrian/hie-bios: streamingProcess: exec: invalid argument (Bad file descriptor)` is shown in the diagnostic. Cannot auto complete, or linting.
it is noted that ghc/compiler/hadrain/hie-bios does not exist.
using hls through nvim-lspconfig work.   

### The minimal config used to reproduce this issue.

```Lua
default config
```

**Comment by mrcjkb** [comment]
Hey :wave: 

Moving this to Discussions, as this is not a haskell-tools bug. It's an error message shown by haskell-language server.

> using hls through nvim-lspconfig work

- Are you using mason or mason-lspconfig?
- Can you confirm that you are using the same version/build of haskell-language-server in both cases? (If you're using mason, then probably not).

> How to reproduce the issue
> `default config`

> The minimal config used to reproduce this issue.
> `default config`

A note for future reference: I use this plugin almost daily, and do not observe this behaviour, so "default config" is not very helpful for me to reproduce issues.

Please see the [troubleshooting section](https://github.com/mrcjkb/haskell-tools.nvim?tab=readme-ov-file#troubleshooting) in the readme.
Specifically, the [minimal config subsection](https://github.com/mrcjkb/haskell-tools.nvim?tab=readme-ov-file#minimal-config).
